<launch>

<!-- Read Required Command Line Arguments: -->
<arg name="initial_pose" default="$(find zeta_rescue)/data/initial_pose.yaml"/>
<arg name="map_file" />
<arg name="landmark_file" default="$(find zeta_rescue)/data/landmark_file.csv"/>


<!-- Camera marker args -->
<arg name="marker_size" default="6" />
<arg name="max_new_marker_error" default="0.08" />
<arg name="max_track_error" default="0.2" />
<arg name="cam_image_topic" default="/camera/rgb/image_rect_color" />
<arg name="cam_info_topic" default="/camera/rgb/camera_info" />
<arg name="output_frame" default="/camera_rgb_frame" />

<node name="ar_track_alvar" pkg="ar_track_alvar"
      type="individualMarkersNoKinect" respawn="false" output="screen"
      args="$(arg marker_size) $(arg max_new_marker_error) $(arg
      max_track_error) $(arg cam_image_topic) $(arg cam_info_topic)
      $(arg output_frame)" />

<!-- Launch minimal, rviz, amcl navigation -->
<include file="$(find turtlebot_bringup)/launch/minimal.launch" />
<!-- <include file="$(find turtlebot_navigation)/launch/amcl_demo.launch"> -->
<!-- <include file="$(find zeta_rescue)/launch/amcl_zeta.launch"> -->





<!-- Load initial pose parameters from yaml file: -->
<rosparam command="load" file="$(arg initial_pose)" />
<rosparam command="load" file="$(arg map_file)" />


<node name="report_button" pkg="zeta_rescue" type="button.py" />
<!-- <node name="Detector_Alvar" pkg="zeta_rescue" type="detect_alvar.py" /> -->

<include file="$(find turtlebot_rviz_launchers)/launch/view_navigation.launch" />

<!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find zeta_rescue)/rviz/kinect_alvar_demo.rviz" /> -->






  <include file="$(find turtlebot_bringup)/launch/3dsensor.launch">
    <arg name="rgb_processing" value="false" />
    <arg name="depth_registration" value="false" />
    <arg name="depth_processing" value="false" />

    <!-- We must specify an absolute topic name because if not it will be prefixed by "$(arg camera)".
         Probably is a bug in the nodelet manager: https://github.com/ros/nodelet_core/issues/7 --> 
    <arg name="scan_topic" value="/scan" />
  </include>

  <!-- Map server -->

<node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />

  <include file="$(find zeta_rescue)/launch/amcl.launch.xml">
    <!-- <arg name="initial_pose" value="$(arg initial_pose)"/> -->
  </include>

  <include file="$(find turtlebot_navigation)/launch/includes/move_base.launch.xml">
    <!--<arg name="initial_pose" value="$(arg initial_pose)"/> -->
  </include>


</launch>
